ci/parse-trivy-results.py

```diff
@@ -0,0 +1,155 @@
+    error = None
+    if not Path("{}.json".format(FLAGS.trivy_file)).is_file():
+        error = ("Please run \n"
+                 "trivy image --format json --light --no-progress -o {}.json "
+                 "gcr.io/cre-tools/ts-bridge \n").format(FLAGS.trivy_file)
+    elif not Path("{}.table".format(FLAGS.trivy_file)).is_file():
+        error = ("Please run \n"
+                 "trivy image --format --light --no-progress -o {}.table "
+                 "gcr.io/cre-tools/ts-bridge \n").format(FLAGS.trivy_file)
+    elif not Path(FLAGS.token_file).is_file():
+        error = ("Please run \n"
+                 "gcloud secrets versions access latest "
+                 "--secret=Ts-bridge-bot-token --format='get(payload.data)' | "
+                 "tr '_-' '/+' | base64 -d > {}").format(FLAGS.token_file)
+    if error:
+        sys.exit(error)
+
+
+def load_results():
+    """ Load the results from Trivy."""
+    trivy_out_json = "{}.json".format(FLAGS.trivy_file)
+    trivy_out_table = "{}.table".format(FLAGS.trivy_file)
+    with open(trivy_out_json) as f:
+        # trivy_result is the json result used for parsing vulnerabilities
+        # Index is 0 because there is only one target built.
+        trivy_result = json.load(f)[0]
+    with open(trivy_out_table) as f:
+        # trivy_table is the results in a more readable table format which can
+        # be included in GitHub issues.
+        trivy_table = f.read()
+    return [trivy_result, trivy_table]
+
+
+def get_severity_list(vulnerabilities):
+    """Filters out all the severities in a given list of vulnerabilities."""
+    severity_list = [v.get("Severity") for v in vulnerabilities]
+    return sorted(set(severity_list))
+
+
+def get_github_repo():
+    """Connects to GitHub API and returns a repo object."""
+    with open(FLAGS.token_file) as f:
+        token = f.read()
+    github = Github(token)
+    return github.get_repo(FLAGS.repo_name)
+
+
+def create_issue(target_name, num_vulnerabilities, severity_list, table):
+    """Creates a Github issue with vulnerabilities as description."""
+    repo = get_github_repo()
+
+    title = ("Vulnerability [{}] found in {}: Images from commit {} cannot be "
+             "released").format(",".join(severity_list), FLAGS.release_tag,
+                                FLAGS.commit_id)
+
+    intro = ("Trivy has detected {} vulnerabilities in your latest "
+             "build. Please correct this issue so the new images can be "
+             "published on Container Registry.\n").format(num_vulnerabilities)
+    body = [intro]
+    body.append("**Cloud Build ID:** {}".format(FLAGS.build_id))
+    body.append("**Commit ID:** {}".format(FLAGS.commit_id))
+    body.append("**Tag:** {}".format(FLAGS.release_tag))
+    body.append("**Target:** {}".format(target_name))
+    body.append("```")
+    body.append(table)
+    body.append("```")
+    body = "\n".join(body)
+
+    new_issue = repo.create_issue(title=title, body=body)
```

Comment:
What happens if creating an issue fails? Ideally we should fail gracefully on it.

Reply:
Added exception handling to address this.